dlib/control/qlearning_abstract.h

```diff
@@ -0,0 +1,205 @@
+// Copyright (C) 2017 Adrián Javaloy (adrian.javaloy@gmail.com)
+// License: Boost Software License   See LICENSE.txt for the full license.
+#undef DLIB_QLEARNING_ABSTRACT_Hh_
+#ifdef DLIB_QLEARNING_ABSTRACT_Hh_
+
+#include "policy_abstract.h"
+#include "model_abstract.h"
+#include <random>
+
+namespace dlib
+{
+    template <
+        typename model_type
+        >
+    class qlearning
+    {
+        /*!
+            REQUIREMENTS ON model_type
+                model_type is an implementation of the model interface declared in
+                  model_abstract.h.
+
+            WHAT THIS OBJECT REPRESENTS
+                This objects is an implementation of the well-known reinforcement learning
```

Comment:
Fix spelling.

Reply:
object**s** and algorithm**s**

Reply:
The text is wrong, it doesn't use any process_samples

Please make sure all the documentation is correct.  I am extremely anal about having correct and clear documentation :).  You should review all the comments a few times to make sure they are correct and clear before the next PR review.